### PR TITLE
Implement json serialization/deserialization for LSP json messages. Part 1/N.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This PR is just the addition of the types.  The next PR (https://github.com/dotnet/roslyn/pull/68990) is the consumption of these types.

### CODE REVIEW

**Review by dibarbet — COMMENTED** [review]
lgtm (pending final decision)

**Review by arkalyanms — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Moving forward.  Speedometer showed no issues.

File: src/Features/LanguageServer/Protocol/LanguageServerProtocolResources.Designer.cs
Comment (sharwell) [inline src/Features/LanguageServer/Protocol/LanguageServerProtocolResources.Designer.cs:11]: 😕 why are we adding this instead of generating it like all the others?

**Pull request merged**
